crates/nostr/src/nips/nip38.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022-2023 Yuki Kishimoto
+// Copyright (c) 2023-2025 Rust Nostr Developers
+// Distributed under the MIT software license
+
+//! NIP38: User Statuses
+//!
+//! <https://github.com/nostr-protocol/nips/blob/master/38.md>
+
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use core::fmt;
+
+use crate::{Tag, TagStandard, Timestamp};
+
+/// NIP38 types
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum StatusType {
+    /// General status: "Working", "Hiking", etc.
+    General,
+    /// Music what you are currently listening to
+    Music,
+    /// Custom status: "Playing", "Reading", etc.
+    Custom(String),
+}
+
+impl Default for StatusType {
+    fn default() -> Self {
+        Self::General
+    }
+}
+
+impl fmt::Display for StatusType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::General => write!(f, "general"),
+            Self::Music => write!(f, "music"),
+            Self::Custom(s) => write!(f, "{}", s),
+        }
+    }
+}
+
+/// User status
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct LiveStatus {
+    /// Status type, includes: General, Music or Custom
+    pub status_type: StatusType,
+    /// Expiration time of the status (Optional)
+    pub expiration: Option<Timestamp>,
+    /// Reference to the external resource (Optional)
+    pub reference: Option<String>,
+}
+
```

Comment:
What do you thin about adding a constructor to build a status without optional stuff? Like:

```rust
impl LiveStatus {
    pub fn new(status_type: StatusType) -> Self {
        Self {
            status_type,
            expiration: None,
            reference: None
        }
    }
}
```